lib/ansible/modules/network/nxos/nxos_igmp_interface.py

```diff
@@ -497,7 +497,7 @@ def main():
         oif_routemap=dict(required=False, type='str'),
         oif_prefix=dict(required=False, type='str', removed_in_version='2.10'),
         oif_source=dict(required=False, type='str', removed_in_version='2.10'),
-        oif_ps=dict(required=False, type='list', elements='dict'),
+        oif_ps=dict(required=False, type='raw'),
```

Comment:
```
    oif_ps:
        description:
            - Configure prefixes and sources for static outgoing interface (OIF). This
              is a list of dict where each dict has source and prefix defined or just
              prefix if source is not needed. The specified values will be configured
              on the device and if any previous prefix/sources exist, they will be removed.
              Keyword 'default' is also accpted which removes all existing prefix/sources.

```
Based on the option description, changing type=`list` to type=`raw` might not be the best of the fix. Since in addtion to `dict` as list elements it can accept a special str value `default` the option can be defined as `oif_ps=dict(required=False, type='list')`. This won't check the type of each element, later in the code if the value of `oif_ps` is assumed as `dict`, it might result in error, to avoid that if element value is not `['default']` it can be converted to list dict at line 598
```
 if oif_ps:
        if oif_ps == ['default']:
            delta['oif_ps'] = []
        else:
            delta['oif_ps'] = <convert oif_ps to list of dict>
```

Reply:
Removed this commit from this PR until we decide this on devel https://github.com/ansible/ansible/pull/53136